src/coreclr/vm/jitinterfacegen.cpp

```diff
@@ -38,7 +38,15 @@ void InitJITAllocationHelpers()
             SetJitHelperFunction(CORINFO_HELP_NEWARR_1_ALIGN8, RhpNewArrayFastAlign8);
 #endif
 
-            ECall::DynamicallyAssignFCallImpl(GetEEFuncEntryPoint(RhNewString), ECall::FastAllocateString);
+            PCODE pfnHelper = GetEEFuncEntryPoint(RhNewString);
+
+#ifdef FEATURE_PORTABLE_ENTRYPOINTS
+            AllocMemHolder<PortableEntryPoint> portableEntryPoint;
+            pfnHelper = PortableEntryPoint::CreateFromNativeEntryPoint(pfnHelper, &portableEntryPoint);
+            portableEntryPoint.SuppressRelease();
+#endif // FEATURE_PORTABLE_ENTRYPOINTS
+
+            ECall::DynamicallyAssignFCallImpl(pfnHelper, ECall::FastAllocateString);
```

Comment:
`RhNewString` is the first native FCall we will hit when `InvokeManagedMethod` is called.

Reply:
This is great, I was chasing this one, but missed this place. Do we need to do it for other fcalls too or is the string constructor special?

Reply:
The best approach here is to look for `ECall::DynamicallyAssignFCallImpl()` for these sorts of issue. I think string constructors are the only remaining issues here. I tend to avoid trying to proactively fixing these until we hit them in a bug.